fix: wait_timeout update for the best practices (#16453)
pingcap/docs: develop/dev-guide-connection-parameters.md

### Probe configuration

> The connection pool maintains persistent connections to TiDB, if the version of TiDB:

Reviewer: ```suggestion
The connection pool maintains persistent connections from clients to TiDB as follows:
```

> - Less than `v5.4`, does not proactively close client connections by default (unless an error is reported);
> - Equal or above than `v5.4`, it defaults to closing client connections after `28800` seconds of inactivity, which means `8` hours. You can control this timeout setting using the TiDB and MySQL compatible `wait_timeout` variable, see the [JDBC Query Timeout](/develop/dev-guide-timeouts-in-tidb.md#jdbc-query-timeout) document for details.

Reviewer: ```suggestion
- Before v5.4, TiDB does not proactively close client connections by default (unless an error is reported);
- Starting from v5.4, TiDB automatically closes client connections after `28800` seconds (this is, `8` hours) of inactivity by default. You can control this timeout setting using the TiDB and MySQL compatible `wait_timeout` variable. For more information, see [JDBC Query Timeout](/develop/dev-guide-timeouts-in-tidb.md#jdbc-query-timeout).
```

> 

Reviewer: ```suggestion

Moreover, there might be network proxies such as [LVS](https://en.wikipedia.org/wiki/Linux_Virtual_Server) or [HAProxy](https://en.wikipedia.org/wiki/HAProxy) between clients and TiDB. These proxies typically proactively clean up connections after a specific idle period (determined by the proxy's idle configuration). In addition to monitoring the proxy's idle configuration, connection pools also need to maintain or probe connections for keep-alive.

```

### Probe configuration

The connection pool maintains persistent connections from clients to TiDB as follows:

> - Before v5.4, TiDB does not proactively close client connections by default (unless an error is reported);

Reviewer: ```suggestion
- Before v5.4, TiDB does not proactively close client connections by default (unless an error is reported).
```
